modules/teadsBidAdapter.js

```diff
@@ -1,12 +1,16 @@
 import {getValue, logError, deepAccess, parseSizesInput, isArray, getBidIdParameter} from '../src/utils.js';
 import {registerBidder} from '../src/adapters/bidderFactory.js';
 import {getStorageManager} from '../src/storageManager.js';
+import {startDetection, isAutoplayEnabled} from '../libraries/autoplayDetection/autoplay.js';
 
 /**
  * @typedef {import('../src/adapters/bidderFactory.js').BidRequest} BidRequest
  * @typedef {import('../src/adapters/bidderFactory.js').Bid} Bid
  */
 
+// start autoplay detection as soon as Prebid is loading
+startDetection();
```

Comment:
why not call this in `autoplay.js` ? Every module using this library will need to do it, so might as well do it in the library. I can't think of a case where a module would want to delay this

Reply:
Yes good point, this makes sense since libraries are not included by default, so the detection will only run if a module includes it. I'll make the change!

Reply:
By doing that, we can also remove the `detectionInProgress` logic, since the code is executed once when the module is initialized.

Reply:
Done in ed3427c